core/src/main/java/fj/function/Booleans.java

```diff
@@ -92,6 +84,56 @@ public static boolean and(final List<Boolean> l) {
     return Monoid.conjunctionMonoid.sumLeft(l);
   }
 
+    /**
+     * maps given function to the predicate function
+     * @param p predicate to be mapped over
+     * @param f function
+     * @return predicate function
+     */
+    public static <A, B>  F<B, Boolean> contramap(F<B, A> f, F<A, Boolean> p){
+        return compose(p, f);
+    }
+
```

Comment:
I see now, it is contramapping over the predicate, whereas I was thinking it was mapping over the first function.  It looks like these two cases are equivalent here, but given we are focused on predicates contramapping is appropriate